### PR TITLE
Code simplifications w.r.t mesh->data().at(ID)

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
## Main changes of this PR

Code simplifications.

## Motivation and additional information

Reduces code complexity.

## Author's checklist

* [x] I added a changelog file with this PR number in `docs/changelog/` if there are noteworthy changes.
* [x] I ran `tools/formatting/check-format` and everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.10.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Do you understand the code changes?
* [ ] (more questions/tasks)

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
There seems to be a problem with some of the tests that I will have to investigate. @fsimonis: A thorough review does not make sense at this point in time, but do you agree with the overall direction this PR is heading into?

File: src/cplscheme/config/CouplingSchemeConfiguration.cpp [hunk 1]
```diff
@@ -755,11 +755,7 @@ mesh::PtrData CouplingSchemeConfiguration::findDataByID(
     int ID) const
 {
   for (const mesh::PtrMesh &mesh : _meshConfig->meshes()) {
-    for (mesh::PtrData data : mesh->data()) {
-      if (data->getID() == ID) {
-        return data;
-      }
-    }
+    return mesh->data(ID);
```
Comment (fsimonis) [inline src/cplscheme/config/CouplingSchemeConfiguration.cpp:758]: This requires ID to exist.

**Review by fsimonis — CHANGES_REQUESTED** [review]

File: src/cplscheme/config/CouplingSchemeConfiguration.cpp (hunk 1 above)
Comment (BenjaminRodenberg) [reply src/cplscheme/config/CouplingSchemeConfiguration.cpp:758]: Good point. Will do this.

Comment (BenjaminRodenberg) [reply src/cplscheme/config/CouplingSchemeConfiguration.cpp]: Done in 5ec6d2a761aeb1a5f6c81a65fde85f072eabe237

**Review by fsimonis — APPROVED** [review]

**Pull request merged**